test/e2e/cluster_autoscaler.go

```diff
@@ -0,0 +1,211 @@
+		input    ClusterAutoscalerInstallInput
+	)
+
+	Expect(ctx).NotTo(BeNil(), "ctx is required for %s spec", specName)
+
+	input = inputGetter()
+	Expect(input.BootstrapClusterProxy).ToNot(BeNil(), "Invalid argument. input.BootstrapClusterProxy can't be nil when calling %s spec", specName)
+	Expect(input.Namespace).ToNot(BeNil(), "Invalid argument. input.Namespace can't be nil when calling %s spec", specName)
+	Expect(input.ClusterName).ToNot(BeEmpty(), "Invalid argument. input.ClusterName can't be empty when calling %s spec", specName)
+
+	mgmtClient := bootstrapClusterProxy.GetClient()
+	Expect(mgmtClient).NotTo(BeNil())
+	mdList := framework.GetMachineDeploymentsByCluster(ctx, framework.GetMachineDeploymentsByClusterInput{
+		Lister:      mgmtClient,
+		ClusterName: input.ClusterName,
+		Namespace:   input.Namespace.Name,
+	})
+	var hasClusterAutoscalerConfig bool
+	for _, md := range mdList {
+		if _, ok := md.GetAnnotations()["cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size"]; ok {
+			if _, ok := md.GetAnnotations()["cluster.x-k8s.io/cluster-api-autoscaler-node-group-max-size"]; ok {
+				hasClusterAutoscalerConfig = true
+			}
+		}
+	}
+	if hasClusterAutoscalerConfig {
+		options := &helmVals.Options{
+			Values: []string{
+				"cloudProvider=clusterapi",
+				fmt.Sprintf("autoDiscovery.clusterName=%s", input.ClusterName),
+				fmt.Sprintf("clusterAPIKubeconfigSecret=%s-kubeconfig", input.ClusterName),
+				"clusterAPIMode=kubeconfig-incluster",
+				"extraArgs.scan-interval=1m",
+				"extraArgs.balance-similar-node-groups=true",
+			},
+		}
+		InstallHelmChart(ctx, bootstrapClusterProxy, input.Namespace.Name, clusterAutoscalerHelmRepoURL, clusterAutoscalerChartName, fmt.Sprintf("%s-%s", clusterAutoscalerChartName, input.ClusterName), options, "")
+	}
+	var d = &appsv1.DeploymentList{}
+	Eventually(func(g Gomega) bool {
+		Logf("Listing deployments in namespace %s with label %s=%s", input.Namespace.Name, clusterAutoscalerDeploymentLabelKey, clusterAutoscalerDeploymentLabelVal)
+		err := mgmtClient.List(ctx, d, client.InNamespace(input.Namespace.Name), client.MatchingLabels{clusterAutoscalerDeploymentLabelKey: clusterAutoscalerDeploymentLabelVal})
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(d).NotTo(BeNil())
+		g.Expect(len(d.Items)).To(Equal(1))
+		for _, c := range d.Items[0].Status.Conditions {
+			if c.Type == appsv1.DeploymentAvailable && c.Status == corev1.ConditionTrue {
+				return true
+			}
+		}
+		return false
+	}, e2eConfig.GetIntervals(specName, "wait-deployment")...).Should(BeTrue())
+}
+
+// AzureAutoscalerSpecInput is the input for AzureAutoscalerSpec.
+type AzureAutoscalerSpecInput struct {
+	BootstrapClusterProxy framework.ClusterProxy
+	Namespace             *corev1.Namespace
+	ClusterName           string
+}
+
+// AzureAutoscalerSpec implements a test that verifies VM extensions are created and deleted.
+func AzureAutoscalerSpec(ctx context.Context, inputGetter func() AzureAutoscalerSpecInput) {
```

Comment:
should this test live in the cluster autoscaler project itself to test the CAPI provider? Ideally this would be able to run with multiple CAPI providers, not just Azure, and even in the future could get used for gating PRs (maybe with the Docker provider)

Seems like this is trying to test cluster-autoscaler functionality, not CAPZ functionality